coinflip/main.c

```diff
@@ -0,0 +1,394 @@
+        if(coin_flip==0) {
+            // Go through each of the capsense values to determine when the buttons have been pressed
+            for (int i = 0; i < 4; i++) {
+                // Use the sum of their values to determine if a button has been pressed
+                sum += g_channel_values[i];
+            }
+            // Minimum threshold for the button press to register
+            if(sum > CAPSENSE_DETECT_MIN) {
+                // Use the odd/even value of the tick count to determine if the coin flip was red(head) or green(tails)
+                if(tick_count%2==0){
+                    // Tick count was even, thus turn on the red LED
+                    gpio_clear(LED_RED_PORT, LED_RED_PIN);
+                    coin_flip=1;
+                } else {
+                    // Tick count was odd, thus turn on the green LED
+                    gpio_clear(LED_GREEN_PORT, LED_GREEN_PIN);
+                    coin_flip=2;
+                }
+                // Reset the tick count to use as a timer
+                tick_count = 0;
+            }
+            // Reset the sum value so it doesn't grow across multiple loops
+            sum = 0;
+        }
+
+        // Increment the tick counter so that LED timing can be preserved
+        // an improvement would be to increment this in a system interrupt
+        tick_count++;
+
+        // If the tick count has equaled the appropriate delay for the on time then turn off the LEDs
+        if(tick_count == LED_ON_DELAY) {
+            // Turn off the LEDs
+            gpio_set(LED_GREEN_PORT, LED_GREEN_PIN);
+            gpio_set(LED_RED_PORT, LED_RED_PIN);
+        } else if(tick_count >= LED_OFF_DELAY) {
+            // Once the tick count delay has exceeded, reset the coin_flip variable
+            coin_flip = 0;
+        }
+    }
+}
+
+static void ACMP_CapsenseChannelSet(uint32_t channel)
+{
+    g_current_channel = channel;
+
+    if (channel == 0) {
+        MMIO32(ACMP0_INPUTSEL) = (acmpResistor0 << _ACMP_INPUTSEL_CSRESSEL_SHIFT)
+                        | ACMP_INPUTSEL_CSRESEN
+                        | (false << _ACMP_INPUTSEL_LPREF_SHIFT)
+                        | (0x3f << _ACMP_INPUTSEL_VDDLEVEL_SHIFT) // 0x3f for channel 0 and 0x3d for channel 1
+                        | ACMP_INPUTSEL_NEGSEL(ACMP_INPUTSEL_NEGSEL_CAPSENSE)
+                        | (channel << _ACMP_INPUTSEL_POSSEL_SHIFT);
+    }
+    else if (channel == 1) {
+        MMIO32(ACMP0_INPUTSEL) = (acmpResistor0 << _ACMP_INPUTSEL_CSRESSEL_SHIFT)
+                        | ACMP_INPUTSEL_CSRESEN
+                        | (false << _ACMP_INPUTSEL_LPREF_SHIFT)
+                        | (0x3d << _ACMP_INPUTSEL_VDDLEVEL_SHIFT) // 0x3f for channel 0 and 0x3d for channel 1
+                        | ACMP_INPUTSEL_NEGSEL(ACMP_INPUTSEL_NEGSEL_CAPSENSE)
+                        | (channel << _ACMP_INPUTSEL_POSSEL_SHIFT);
+    }
+    else if (channel == 2) {};
```

Comment:
Did this compile for you? I've had to remove the semicolon after these two lines to compile ('else' without a previous 'if').

Reply:
Not sure when I added those, but have just been in "cleaning up mode".  Surely wouldn't have compiled on my system either.